src/Stratis.Bitcoin.Features.Consensus/Behaviors/ProvenHeadersConsensusManagerBehavior.cs

```diff
@@ -104,6 +104,29 @@ protected override async Task OnMessageReceivedAsync(INetworkPeer peer, Incoming
             }
         }
 
+        /// <inheritdoc />
+        protected override bool CanConsumeCache()
+        {
+            int height = this.consensusManager.Tip.Height;
+
+            // Try cashe consumption every N blocks advanced by consensus.
+            // N should be between 0 and max reorg. 20% of max reorg is a good random value.
+            // Higher the N the better performance boost we can get.
+            uint tryCacheEveryBlocksCount = this.network.Consensus.MaxReorgLength / 5;
```

Comment:
this calculation can go inside the if bellow

Reply:
Fixed